tests/test_user_security.py

```diff
@@ -0,0 +1,200 @@
+user_data_access = UserDataAccess()
+
+# SECURITY HANDLER REGISTER
+key = 'super-secret'
+security_handler = CustomerSecurity('user_cookie', key, timedelta(seconds=2))
+SecurityHandler.register(security_handler)
+
+# DJANGO INIT
+if not settings.configured:
+    settings.configure()
+    settings.DEFAULT_CHARSET = 'utf-8'
+web_util = WebHelpersUtils()
+web_util.init(web_util.ServerType.DJANGO)
+
+
+class TestUserSecurity(unittest.TestCase):
+
+    user = {'id': 1, 'email': 'user@def.com'}
+    tester = {'id': 2, 'email': 'tester@def.com'}
+    admin = {'id': 3, 'email': 'admin@def.com'}
+
+    @classmethod
+    def setUp(cls):
+        user_data_access.create({'username': 'user', 'email': 'user@def.com', 'role': PolicyRoles.USER.value})
+        user_data_access.create({'username': 'tester', 'email': 'tester@def.com', 'role': PolicyRoles.TESTER.value})
+        user_data_access.create({'username': 'admin', 'email': 'admin@def.com', 'role': PolicyRoles.ADMIN.value})
+
+    def _create_request(self, payload: dict):
+        request_object = HttpRequest
+        token = jwt.encode(payload, key)
+        request_object.COOKIES = {'user_cookie': token}
+        return request_object
+
+    def test_secure_entry(self):
+
+        response = user_entry(self._create_request(TestUserSecurity.user))
+        self.assertEqual(response.status_code, 200)
+
+        response = tester_entry(self._create_request(TestUserSecurity.tester))
+        self.assertEqual(response.status_code, 200)
+
+        response = tester_admin_entry(self._create_request(TestUserSecurity.tester))
+        self.assertEqual(response.status_code, 200)
+
+        response = admin_entry(self._create_request(TestUserSecurity.admin))
+        self.assertEqual(response.status_code, 200)
+
+        response = tester_admin_entry(self._create_request(TestUserSecurity.admin))
+        self.assertEqual(response.status_code, 200)
+
+        response = user_entry(self._create_request(TestUserSecurity.admin))
+        self.assertEqual(response.status_code, 401)
+
+        response = tester_entry(self._create_request(TestUserSecurity.admin))
+        self.assertEqual(response.status_code, 401)
+
+        response = admin_entry(self._create_request(TestUserSecurity.user))
+        self.assertEqual(response.status_code, 401)
+
+        response = tester_entry(self._create_request(TestUserSecurity.user))
+        self.assertEqual(response.status_code, 401)
+
+        response = tester_admin_entry(self._create_request(TestUserSecurity.user))
+        self.assertEqual(response.status_code, 401)
+
+        response = admin_entry(self._create_request(TestUserSecurity.tester))
+        self.assertEqual(response.status_code, 401)
+
+        response = user_entry(self._create_request(TestUserSecurity.tester))
+        self.assertEqual(response.status_code, 401)
+
+        response = user_entry(self._create_request({'id': 3, 'email': 'user@def.com'}))
+        self.assertEqual(response.status_code, 403)
+

```

Comment:
add 2 tests cases 
1. request that have access 
2. freeze time
3. same request as in 1 , but expect it to fail